src/ImageRunModal.jsx

```diff
@@ -632,6 +587,79 @@
         return owner === systemOwner;
     };
 
+    isFormValid = (validationFailed) => {
+        return Object.keys(validationFailed).length == 0;
+    };
+
+    async validateContainerName(containerName) {
+        try {
+            await client.containerExists(this.isSystem(), containerName);
+        } catch (error) {
+            return;
+        }
+        return _("Name already in use");
+    }
+
+    async validateForm() {
+        const { publish, volumes, env, containerName } = this.state;
+        const validationFailed = { };
+
+        const publishValidation = publish.map(a => {
+            let publishPortValidation;
+            publishPortValidation = validatePublishPort(a, "IP", publishPortValidation);
+            publishPortValidation = validatePublishPort(a, "hostPort", publishPortValidation);
+            publishPortValidation = validatePublishPort(a, "containerPort", publishPortValidation);
+
+            return publishPortValidation;
+        });
+        if (Object.keys(publishValidation).length > 0)
+            validationFailed.publish = publishValidation;
+
+        const volumesValidation = volumes.map(a => {
+            let volumesValidation;
+            volumesValidation = validateVolume(a, "hostPath", volumesValidation);
+            volumesValidation = validateVolume(a, "containerPath", volumesValidation);
+
+            return volumesValidation;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230907-093411-a9f37bd4-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)